tsrc/workspace/__init__.py

```diff
@@ -71,12 +71,17 @@ def clone_missing(self) -> None:
             repo_path = self.root_path / repo.src
             if not repo_path.exists():
                 to_clone.append(repo)
-        cloner = Cloner(self.root_path, shallow=self.config.shallow_clones)
+        cloner = Cloner(
+            self.root_path,
+            shallow=self.config.shallow_clones,
+            remote_name=self.config.singular_remote,
+        )
         tsrc.executor.run_sequence(to_clone, cloner)
 
     def set_remotes(self) -> None:
-        remote_setter = RemoteSetter(self.root_path)
-        tsrc.executor.run_sequence(self.get_repos(), remote_setter)
+        if self.config.singular_remote is None:
```

Comment:
Better : `if not self.config.singular_remote`

Reply:
Argh, yes, that was spec'd in the code manifesto. It slipped my mind. I also fixed two other places in the code where I didn't rely on `None` being truth-y.